### DISCUSSION TITLE
regenerating *.i file from *_outfile.e file

### DISCUSSION THREAD

**Discussion by DomenicoCFD** [idaholab/moose#18789] [Q&A General] [answered]
Hi All,

I would like to regenerate the input file of MOOSE if only the _outfile.e is given. This is because I would expect to modify the input file and therefore I may well loose the track of all the input that I provided in MOOSE for a particular _outfile.e MOOSE result file. 

A similar feature does exist in ANSYS so eventually MOOSE does have a similar thing.

I would really appreciate your help on this.

Kind Regards,

**Comment by dschwen** [comment, marked as answer]
The input file is actually stored in the exodus output as a data block. You can reveal some of it on the command line using `strings my_out.e`. Note that this is a reconstructed input with all default parameters inserted by MOOSE.

**Reply by dschwen** [reply]
You can extract the text properly with the `ncdump` command from netcdf (the Exodus II fileformat is based on netcdf)

**Reply by dschwen** [reply]
`ncdump -v info_records my_out.e`

**Reply by DomenicoCFD** [reply]
Hi,

Many thanks for your direction. Following your instruction, it seems that the "ncdump" option does give a better format than "strings". Other than that, is there a command line which returns the .i file ready to be rerun in MOOSE?

I look forward to hearing from you.

Kind Regards,

Domenico

**Reply by dschwen** [reply]
No single command. I suppose we could write a python script though. There is likely some post processing required though before passing the extracted input back to moose. Things like

```
inactive                     = (no_default)
```

or 

```
  "    constant_properties          = \'1 3.17e+07 0.32 0 3.17e+07 0.32 1000 6.67...",
  "e-06 100 1e-08 5000 4 0.02 0\'",
```

(which uses `...` to mark a broken string)

need special treatment.

**Reply by aaelmeli** [reply]
Hi
you can check Matlab ncread(), see this discussion #17852.
